### DISCUSSION TITLE
How can I add custom (downloaded) filters

### DISCUSSION THREAD

**Discussion by surajpunemaha** [burhanrashid52/PhotoEditor#323] [Q&A]
I want to add filters which my client will provide. Is there any way to add that? if can add, what should be the file format for filters?

**Comment by burhanrashid52** [comment]
There is no direct way to get a filter. You can use [EffectFactory](https://developer.android.com/reference/android/media/effect/EffectFactory) to get different filters. The library has a predefined set of EffectFactory filter, you can create use [CustomeEffect](https://github.com/burhanrashid52/PhotoEditor/wiki/Filter-Effect#custom-filter) API to create custom filters.

**Comment by surajpunemaha** [comment]
I checked it. I want to apply different filters from EffectFactory at once. Is it possible using this library?
for example apply MyCustomEffect = EFFECT_BRIGHTNESS + EFFECT_CROP + EFFECT_FLIP
more effects in single instance. I don't want to save state or intermediate results in bitmap format.
Other suggestions are also welcomed.
